extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolProducer.java

```diff
@@ -4,36 +4,87 @@
 import javax.enterprise.inject.Produces;
 import javax.inject.Singleton;
 
+import org.jboss.logging.Logger;
+
 import io.vertx.pgclient.PgPool;
 
 @ApplicationScoped
 public class PgPoolProducer {
 
-    private volatile PgPool pgPool;
```

Comment:
I believe the `volatile` construct was preferred here because the producer can be invoked concurrently at runtime (e.g. in request scoped beans). Using synchronization instead would serialize these calls. Or am I missing something?

Reply:
I actually want the serialization here to catch the creation of the deprecated clients. 
I can restore using a volatile for the `PgPool` but that mix 2 synchronization constructs.

Reply:
Fixed.